public/app/components/fontViewer/fontViewer.js

```diff
@@ -0,0 +1,477 @@
+      var maxHeight = head.yMax - head.yMin;
+      var ctx = glyphBgCanvas.getContext('2d');
+
+      glyphScale = Math.min(
+        glyphW / (head.xMax - head.xMin), glyphH / maxHeight
+      );
+      glyphSize = glyphScale * vm.font.unitsPerEm;
+      glyphBaseline = glyphMargin + glyphH * head.yMax / maxHeight;
+
+      function hline(text, yunits) {
+        var ypx = glyphBaseline - yunits * glyphScale;
+        ctx.fillText(text, 2, ypx + 3);
+        ctx.fillRect(80, ypx, w, 1);
+      }
+
+      ctx.clearRect(0, 0, w, h);
+      ctx.fillStyle = '#a0a0a0';
+      hline('Baseline', 0);
+      // hline('yMax', vm.font.tables.head.yMax);
+      // hline('yMin', vm.font.tables.head.yMin);
+      hline('Ascender', vm.font.tables.hhea.ascender);
+      hline('Descender', vm.font.tables.hhea.descender);
+      // hline('Typo Ascender', vm.font.tables.os2.sTypoAscender);
+      // hline('Typo Descender', vm.font.tables.os2.sTypoDescender);
+    }
+
+    function onFontLoaded() {
+      window.font = vm.font;
+
+      var fontName = vm.font.names.fullName.en;
+      document.getElementById('font-name').innerHTML = fontName;
+
+      var w = cellWidth - cellMarginLeftRight * 2;
+      var h = cellHeight - cellMarginTop - cellMarginBottom;
+      var head = vm.font.tables.head;
+      var maxHeight = head.yMax - head.yMin;
+
+      fontScale = Math.min(w / (head.xMax - head.xMin), h / maxHeight);
+      fontSize = fontScale * vm.font.unitsPerEm;
+      fontBaseline = cellMarginTop + h * head.yMax / maxHeight;
+
+      var pagination = document.getElementsByClassName('pagination')[0];
+      pagination.innerHTML = '';
+      var fragment = document.createDocumentFragment();
+      var numPages = Math.ceil(vm.font.numGlyphs / cellCount);
+      for (var i = 0; i < numPages; i++) {
+        var link = document.createElement('span');
+        var lastIndex = Math.min(
+          vm.font.numGlyphs - 1, (i + 1) * cellCount - 1
+        );
+        link.textContent = i * cellCount + '-' + lastIndex;
+        link.id = 'p' + i;
+        link.addEventListener('click', pageSelect, false);
+        fragment.appendChild(link);
+        // A white space allows to break very long lines into multiple lines.
+        // This is needed for fonts with thousands of glyphs.
+        fragment.appendChild(document.createTextNode(' '));
+      }
+      pagination.appendChild(fragment);
+
+      initGlyphDisplay();
+      displayGlyphPage(0);
+
+      var glyphRange = vm.font.numGlyphs > 100 ? 100 : vm.font.numGlyphs;
+      selectedGlyphIndex = getRandomIntInclusive(0, glyphRange);
+      selectCurrentGlyph(selectedGlyphIndex);
+
+      displayGlyph(selectedGlyphIndex);
+      displayGlyphData(selectedGlyphIndex);
+    }
+
+    function getRandomIntInclusive(min, max) {
+      do {
+        var index = Math.floor(Math.random() * (max - min + 1)) + min;
+        var tempGlyph = vm.font.glyphs.get(index);
+      }
+      while (ignoreChars.indexOf(tempGlyph.name) != -1);
```

Comment:
you can have scenarios where a `glyph`'s name is not defined. you can check our own font in this scenario: `/public/assets/lib/glypher-font/fonts/glypher.ttf`

you can bypass this by changing this to
```javascript
while (tempGlyph.name && ignoreChars.indexOf(tempGlyph.name) != -1);
```